ethereum/contracts/IncentivizedReceiveChannel.sol

```diff
@@ -0,0 +1,173 @@
+pragma solidity >=0.6.2;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "./Decoder.sol";
+
+contract IncentivizedReceiveChannel {
+    using Decoder for bytes;
+
+    uint256 public lastProcessedNonce;
+
+    uint256 public MAX_PAYLOAD_BYTE_SIZE = 1000;
+    uint256 public MAX_PAYLOAD_GAS_COST = 500000;
+    uint256 public EXTERNAL_CALL_COST = 21000;
+    uint256 public MAX_GAS_PER_MESSAGE =
+        EXTERNAL_CALL_COST + MAX_PAYLOAD_BYTE_SIZE + EXTERNAL_CALL_COST;
```

Comment:
Shall we not multiply `MAX_PAYLOAD_BYTE_SIZE` by a `GAS_COST_PER_PAYLOAD_BYTE` here?

Reply:
ye looks like a bug, will fix, though in general fee/incentive/gas constraints are planned to do properly in follow up milestone/PR

Reply:
noted in https://github.com/Snowfork/polkadot-ethereum/pull/250